clippy_lints/src/methods/iter_skip_zero.rs

```diff
@@ -0,0 +1,36 @@
+use clippy_utils::{
+    consts::{constant, Constant},
+    diagnostics::span_lint_and_sugg,
+    is_from_proc_macro, is_trait_method,
+};
+use rustc_errors::Applicability;
+use rustc_hir::Expr;
+use rustc_lint::LateContext;
+use rustc_span::sym;
+
+use super::ITER_SKIP_ZERO;
+
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'tcx>, arg_expr: &Expr<'_>) {
+    if !expr.span.from_expansion()
+        && is_trait_method(cx, expr, sym::Iterator)
+        && let Some(arg) = constant(cx, cx.typeck_results(), arg_expr).and_then(|constant| {
+            if let Constant::Int(arg) = constant {
+                Some(arg)
+            } else {
+                None
+            }
+        })
+        && arg == 0
+        && !is_from_proc_macro(cx, expr)
+    {
+        span_lint_and_sugg(
+            cx,
+            ITER_SKIP_ZERO,
+            arg_expr.span,
+            "usage of `.skip(0)`",
+            "if you meant to skip the first element, try",
+            "1".to_owned(),
+            Applicability::MaybeIncorrect,
```

Comment:
I'm not sure this should be a suggestion, as it changes the program's logic flow.

Reply:
There are other lints like that, like `almost_complete_range`. I'm fine with changing it but imo just because it changes the program's behavior doesn't mean it shouldn't give a suggestion

Reply:
`MaybeIncorrect` is intended for changes, which change the semantics, but it's likely that the change is preferable.

rust-lang/rustfix#200 might be an interesting read, with more context and also something which could be an interesting project :)